// ng-image-fullscreen-view/esm5/lib/ng-image-fullscreen-view.service.js
import { __decorate, __metadata } from "tslib";
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
var NgImageFullscreenViewService = /** @class */ (function () {
    function NgImageFullscreenViewService() {
    }
    NgImageFullscreenViewService.prototype.isBase64 = function (str) {
        var base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/;
        return base64regex.test(str);
    };
    NgImageFullscreenViewService.prototype.base64FileExtension = function (str) {
        return str.substring('data:image/'.length, str.indexOf(';base64'));
    };
    NgImageFullscreenViewService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgImageFullscreenViewService_Factory() { return new NgImageFullscreenViewService(); }, token: NgImageFullscreenViewService, providedIn: "root" });
    NgImageFullscreenViewService = __decorate([ __metadata("design:paramtypes", [])
    ], NgImageFullscreenViewService);
NgImageFullscreenViewService.ɵfac = function NgImageFullscreenViewService_Factory(t) { return new (t || NgImageFullscreenViewService)(); };
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgImageFullscreenViewService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
    return NgImageFullscreenViewService;
}());
export { NgImageFullscreenViewService };
