// ngx-material-file-input/esm2015/lib/material-file-input.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { FocusMonitor } from '@angular/cdk/a11y';
import { FileInputComponent } from './file-input/file-input.component';
import { ByteFormatPipe } from './pipe/byte-format.pipe';
import * as ɵngcc0 from '@angular/core';
let MaterialFileInputModule = class MaterialFileInputModule {
};
MaterialFileInputModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MaterialFileInputModule });
MaterialFileInputModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MaterialFileInputModule_Factory(t) { return new (t || MaterialFileInputModule)(); }, providers: [FocusMonitor] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MaterialFileInputModule, { declarations: function () { return [FileInputComponent, ByteFormatPipe]; }, exports: function () { return [FileInputComponent, ByteFormatPipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialFileInputModule, [{
        type: NgModule,
        args: [{
                declarations: [FileInputComponent, ByteFormatPipe],
                providers: [FocusMonitor],
                exports: [FileInputComponent, ByteFormatPipe]
            }]
    }], null, null); })();
export { MaterialFileInputModule };
