// ng-image-fullscreen-view/esm2015/lib/ng-image-fullscreen-view.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SliderCustomImageComponent } from './slider-custom-image/slider-custom-image.component';
import { NgImageFullscreenViewComponent } from './ng-image-fullscreen-view.component';
import * as ɵngcc0 from '@angular/core';
let NgImageFullscreenViewModule = class NgImageFullscreenViewModule {
};
NgImageFullscreenViewModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgImageFullscreenViewModule });
NgImageFullscreenViewModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgImageFullscreenViewModule_Factory(t) { return new (t || NgImageFullscreenViewModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgImageFullscreenViewModule, { declarations: function () { return [SliderCustomImageComponent,
        NgImageFullscreenViewComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgImageFullscreenViewComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgImageFullscreenViewModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    SliderCustomImageComponent,
                    NgImageFullscreenViewComponent
                ],
                exports: [
                    NgImageFullscreenViewComponent
                ]
            }]
    }], null, null); })();
export { NgImageFullscreenViewModule };
