// ng-image-fullscreen-view/esm2015/lib/ng-image-fullscreen-view.service.js
import { __decorate, __metadata } from "tslib";
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
let NgImageFullscreenViewService = class NgImageFullscreenViewService {
    constructor() { }
    isBase64(str) {
        const base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/;
        return base64regex.test(str);
    }
    base64FileExtension(str) {
        return str.substring('data:image/'.length, str.indexOf(';base64'));
    }
};
NgImageFullscreenViewService.ɵfac = function NgImageFullscreenViewService_Factory(t) { return new (t || NgImageFullscreenViewService)(); };
NgImageFullscreenViewService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgImageFullscreenViewService_Factory() { return new NgImageFullscreenViewService(); }, token: NgImageFullscreenViewService, providedIn: "root" });
NgImageFullscreenViewService = __decorate([ __metadata("design:paramtypes", [])
], NgImageFullscreenViewService);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgImageFullscreenViewService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
export { NgImageFullscreenViewService };
